processing/src/main/java/io/druid/query/groupby/GroupByQueryHelper.java

```diff
@@ -88,13 +92,22 @@ public String apply(DimensionSpec input)
 
     final boolean sortResults = query.getContextValue(CTX_KEY_SORT_RESULTS, true);
 
+    // All groupBy dimensions are strings, for now.
+    final List<DimensionSchema> dimensionSchemas = Lists.newArrayList();
+    for (DimensionSpec dimension : query.getDimensions()) {
+      dimensionSchemas.add(new StringDimensionSchema(dimension.getOutputName()));
+    }
+
+    final IncrementalIndexSchema indexSchema = new IncrementalIndexSchema.Builder()
+        .withDimensionsSpec(new DimensionsSpec(dimensionSchemas, null, null))
+        .withMetrics(aggs.toArray(new AggregatorFactory[aggs.size()]))
```

Comment:
A small thing: `toArray(new T[0])` is a better default choice now, see https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion, "Bottom line" section.